ENIGMAsystem/SHELL/Platforms/SDL/Joystick.cpp

```diff
@@ -77,48 +80,37 @@ double joystick_axis(int id, int numb) {
 namespace enigma {
 
 bool joystick_init() {
-  double init;
-  init = (SDL_InitSubSystem(SDL_INIT_JOYSTICK) > 0);
-  if (joystick1 == NULL)
-    joystick1 = SDL_JoystickOpen(0);
-
-  if (joystick2 == NULL)
-    joystick2 = SDL_JoystickOpen(1);
-
+  for (size_t i = 0; i < joysticks.size(); i++) {
+    joysticks[i] = nullptr;
+  }
+  bool init = (SDL_InitSubSystem(SDL_INIT_JOYSTICK) > 0);
+  joysticks_open();
   return init;
 }
 
 void joystick_uninit() {
-  if (joystick1 != NULL)
-    SDL_JoystickClose(joystick1);
-
-  if (joystick2 != NULL)
-    SDL_JoystickClose(joystick2);
-
+  joysticks_close();
   SDL_QuitSubSystem(SDL_INIT_JOYSTICK);
 }
 
-void joystick_update() {
-  int joystick_count = SDL_NumJoysticks();
-  if (joystick_count <= 0) {
-    if (joystick1 != NULL) {
-      SDL_JoystickClose(joystick1);
-      joystick1 = NULL;
-    }
-
-    if (joystick2 != NULL) {
-      SDL_JoystickClose(joystick2);
-      joystick2 = NULL;
+void joysticks_open() {
```

Comment:
This will work for first run but as soon as you start plugging / unplugging things will get messy. You should be passing the joystick id from the event and using that to call open/close. 

https://github.com/enigma-dev/enigma-dev/blob/master/ENIGMAsystem/SHELL/Platforms/SDL/Gamepad.cpp#L38

Reply:
You mean open and close based on this? https://github.com/enigma-dev/enigma-dev/blob/5881bfb3a6eb53b67d30f2b7ae6fd72211a69ca1/ENIGMAsystem/SHELL/Platforms/SDL/Gamepad.cpp#L55

Reply:
No. I mean look at my logic for adding / removing from gamepads vector. your code should be a mirror of it

Reply:
@fundies I'm not going to test it just yet because there are other things I have to do atm, but feel free to point out anything I probably did wrong until I'm available again to address it.